catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -95,15 +91,45 @@ public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramew
         this.filterBuilder = filterBuilder;
         this.fileWriter = fileWriter;
         this.config = config;
+
+        this.taskManager = new MigrationTaskManager(config, fileWriter);
+    }
+
+    /**
+     * Constructor that allows the caller choice of {@link MigrationTaskManager} to be injected.
+     *
+     * @param description   description of this migratable
+     * @param framework     framework used to retrieve the metacards to export
+     * @param filterBuilder builder used to create query filters
+     * @param fileWriter    object used to write metacards to disk
+     * @param config        export configuration information
+     * @param taskManager   the manager responsible for submitting file writing jobs during export
+     */
+    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+            @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
+            @NotNull CatalogMigratableConfig config, @NotNull MigrationTaskManager taskManager) {
+        super(description, true);
+
+        notNull(framework, "CatalogFramework cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+        notNull(taskManager, "Task manager cannot be null");
+
+        this.framework = framework;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+
+        this.taskManager = taskManager;
     }
 
     /**
      * Exports all the metacards currently stored in the catalog framework.
-     * <p/>
+     * <p>
      * {@inheritDoc}
      */
     public MigrationMetadata export(Path exportPath) throws MigrationException {
```

Comment:
This was later discussed as a separate matter. Instead of trying to synchronize individual `migratables` we will defer it to the `ConfigurationMigrationManager`.